eslint-configs/eslint-config-seekingalpha-typescript/rules/typescript-eslint/index.js

```diff
@@ -0,0 +1,389 @@
+module.exports = {
+  plugins: ['@typescript-eslint/eslint-plugin'],
+
+  rules: {
+
+    'import/extensions': [
+      'error',
+      'ignorePackages',
+      {
+        js: 'never',
+        jsx: 'never',
+        ts: 'never',
+        tsx: 'never',
+      },
+    ],
+
+    '@typescript-eslint/adjacent-overload-signatures': 'error',
+
+    '@typescript-eslint/await-thenable': 'error',
+
+    '@typescript-eslint/ban-ts-comment': 'error',
+
+    '@typescript-eslint/ban-tslint-comment': 'error',
+
+    '@typescript-eslint/ban-types': 'error',
+
+    'brace-style': 'off',
+    '@typescript-eslint/brace-style': 'error',
+
+    '@typescript-eslint/class-literal-property-style': 'error',
+
+    'comma-dangle': 'off',
+    '@typescript-eslint/comma-dangle': [
+      'error', {
+        arrays: 'always-multiline',
+        objects: 'always-multiline',
+        imports: 'always-multiline',
+        exports: 'always-multiline',
+        functions: 'never',
+      },
+    ],
+
+    'comma-spacing': 'off',
+    '@typescript-eslint/comma-spacing': 'error',
+
+    '@typescript-eslint/consistent-generic-constructors': 'error',
+
+    '@typescript-eslint/consistent-indexed-object-style': 'error',
+
+    '@typescript-eslint/consistent-type-assertions': 'error',
+
+    '@typescript-eslint/consistent-type-definitions': 'error',
+
+    '@typescript-eslint/consistent-type-exports': 'error',
+
+    '@typescript-eslint/consistent-type-imports': 'error',
+
+    'default-param-last': 'off',
+    '@typescript-eslint/default-param-last': 'error',
+
+    'dot-notation': 'off',
+    '@typescript-eslint/dot-notation': 'error',
+
+    'func-call-spacing': 'off',
+    '@typescript-eslint/func-call-spacing': 'error',
+
+    'indent': 'off',
+    '@typescript-eslint/indent': [
+      'error',
+      2,
+      {
+        SwitchCase: 1,
+        VariableDeclarator: 1,
+        outerIIFEBody: 1,
+        MemberExpression: 0,
+        FunctionDeclaration: {
+          parameters: 1,
+          body: 1,
+        },
+        FunctionExpression: {
+          parameters: 1,
+          body: 1,
+        },
+        CallExpression: {
+          arguments: 1,
+        },
+        ArrayExpression: 1,
+        ObjectExpression: 1,
+        ImportDeclaration: 1,
+        flatTernaryExpressions: false,
+        offsetTernaryExpressions: false,
+        ignoredNodes: [],
+        ignoreComments: false,
+      },
+    ],
+
+    'init-declarations': 'off',
```

Comment:
all of eslint rules should be in separate file with comment regarding why they are disables